### PR TITLE
[IWT6] Device info component

### PR DESCRIPTION
**Pull request by KrzysztofMoszczynski**
I added the component which displays information about one device in form of extendible block. I used there `react-collapse` library for extendind the data, and `react-switch` for toggle element which is currently not connected to anything. The component has 5 props:
- name
- location
- temperature
- conductivity
- ph

### CODE REVIEW

File: app/javascript/pages/app/components/device-info.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            onHandleColor='#2693e6'
+            handleDiameter={20}
+            uncheckedIcon={false}
+            checkedIcon={false}
+            boxShadow='0px 1px 5px rgba(0, 0, 0, 0.6)'
+            activeBoxShadow='0px 0px 1px 10px rgba(0, 0, 0, 0.2)'
+            height={12}
+            width={30}
+          />
+          {/*possibly to be done by useContext but I wanted it to be visible already*/}
```
Comment (Eleara) [inline app/javascript/pages/app/components/device-info.jsx:36]: This comment is unnecessary.

Comment (Eleara) [inline app/javascript/pages/app/components/device-info.jsx:36]: This comment is unnecessary.

File: app/javascript/pages/app/styles/device-info.scss [hunk 2]
```diff
@@ -0,0 +1,58 @@
+.device-info-wrapper {
+  width: 600px;
+  display: flex;
+  flex-direction: column;
+  background-color: white;
+  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
+  border-radius: 15px;
+  font-family: Poppins;
+  //color: blue;
```
Comment (Eleara) [inline app/javascript/pages/app/styles/device-info.scss:9]: This comment should also be deleted.

File: app/javascript/pages/app/styles/device-info.scss [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      flex-grow: 2;
+      text-align: left;
+      font-size: 18px;
+      color: #de6b48;
+    }
+    .location {
+      font-size: 12px;
+      color: #4f4f4f;
+    }
+    .collapse-arrow {
```
Comment (Eleara) [inline app/javascript/pages/app/styles/device-info.scss:29]: The color of this arrow should be #4f4f4f, as all "black" things in this app. 😄

File: app/javascript/pages/app/icons/collapse_arrow-icon.svg [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (Eleara) [inline app/javascript/pages/app/icons/collapse_arrow-icon.svg:1]: Where did you get this icon from? The svg file looks a bit different than the other ones.

File: app/javascript/pages/app/icons/collapse_arrow-icon.svg [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/icons/collapse_arrow-icon.svg:41]: This file looks funky. All those extra `g` tags could probably be removed. 💁

File: app/javascript/pages/app/styles/device-info.scss (hunk 2 above)
Comment (karczewskiPiotr) [inline app/javascript/pages/app/styles/device-info.scss:9]: `font-family` is set for whole application so this is unnecessary. Also, remove comments.

File: app/javascript/pages/app/components/app.jsx [hunk 6]
```diff
@@ -29,6 +30,13 @@ const App = () => {
               </div>
             </Route>
           </Switch>
+          <DeviceInfo
+            name='Device 666'
+            location='twoja stara'
+            temperature='69'
+            conductivity='sam jesteś kabel'
+            ph='ja mam szczęscie a nie pecha'
+          />
```
Comment (Eleara) [inline app/javascript/pages/app/components/app.jsx:39]: I would suggest changing this to something appropriate.

File: app/javascript/pages/app/styles/device-info.scss [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      width: 15px;
+      height: 15px;
+      cursor: pointer;
+    }
```
Comment (Eleara) [inline app/javascript/pages/app/styles/device-info.scss:33]: The color of this arrow icon should be #4f4f4f, like all "black" things in this app. 😄

File: app/javascript/pages/app/components/device-info.jsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+              Location: <b>{location}</b>
+            </div>
+          ) : (
+            <img
+              src={CollapseArrow}
+              onClick={handleArrowClick}
+              className='collapse-arrow'
+            />
+          )}
+          {isOpened}
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/components/device-info.jsx:51]: What is this doing here? 😕

File: package.json [hunk 9]
```diff
@@ -19,10 +19,14 @@
     "prettier": "^2.0.1",
     "prop-types": "^15.7.2",
     "react": "^16.13.0",
+    "react-collapse": "^5.0.1",
+    "react-collapsible": "^2.7.0",
+    "react-collapsible-component": "^1.3.4",
```
Comment (karczewskiPiotr) [inline package.json:24]: Remove unused packages with yarn. 😠 They will only slow down loading times and take up space.

File: app/javascript/pages/app/components/device-info.jsx (hunk 1 above)
Comment (karczewskiPiotr) [inline app/javascript/pages/app/components/device-info.jsx:36]: Bye, bye comment. 👋

File: app/javascript/pages/app/components/device-info.jsx [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+  };
+
+  const handleSwitchChange = () => {
+    setIsVisible(!isVisible);
+  };
+
+  return (
+    <div className='device-info-wrapper'>
+      <div className='device-info-title'>
+        <div className='device-info-switch'>
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/components/device-info.jsx:22]: Remove this `div`, it's unnecessary.

File: app/javascript/pages/app/components/device-info.jsx [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+            checkedIcon={false}
+            boxShadow='0px 1px 5px rgba(0, 0, 0, 0.6)'
+            activeBoxShadow='0px 0px 1px 10px rgba(0, 0, 0, 0.2)'
+            height={12}
+            width={30}
+          />
+          {/*possibly to be done by useContext but I wanted it to be visible already*/}
+        </div>
+        <div className='device-info-name'>{name}</div>
+        <div>
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/components/device-info.jsx:39]: This `div` isn't needed as well.

File: app/javascript/pages/app/styles/device-info.scss [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+  display: flex;
+  flex-direction: column;
+  background-color: white;
+  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
+  border-radius: 15px;
+  font-family: Poppins;
+  //color: blue;
+  .device-info-title {
+    display: flex;
+    justify-content: space-between;
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/styles/device-info.scss:12]: Not needed.

File: app/javascript/pages/app/styles/device-info.scss [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    align-items: center;
+    padding: 2px 15px 2px 15px;
+    .device-info-switch {
+      padding-top: 7px;
+    }
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/styles/device-info.scss:17]: Remove `.device-info-switch` and add `min-height: 35px` to `device-info-title`.

File: app/javascript/pages/app/styles/device-info.scss [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+    .device-info-name {
+      padding-left: 15px;
+      flex-grow: 2;
+      text-align: left;
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/styles/device-info.scss:21]: Useless I think.

File: app/javascript/pages/app/styles/device-info.scss [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+      color: #4f4f4f;
+    }
+    .collapse-arrow {
+      width: 15px;
+      height: 15px;
+      cursor: pointer;
+    }
+  }
+  .device-info-list {
+    text-align: left;
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/styles/device-info.scss:36]: Useless as well, it seems.

File: app/javascript/pages/app/styles/device-info.scss [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+    list-style-type: none;
+    margin-left: -23px;
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/styles/device-info.scss:38]: Please never do negative margins. Replace this with `padding: 15px`.

File: app/javascript/pages/app/styles/device-info.scss [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+    .list-item-title {
+      font-size: 15px;
+      margin-bottom: -12px;
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/styles/device-info.scss:41]: Change value to `0px`

File: app/javascript/pages/app/styles/device-info.scss [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    .list-item-value {
+      font-size: 12px;
+      color: #4f4f4f;
+    }
```
Comment (karczewskiPiotr) [inline app/javascript/pages/app/styles/device-info.scss:46]: Add `margin: 0px` to this class.

File: app/javascript/pages/app/components/app.jsx (hunk 6 above)
Comment (karczewskiPiotr) [reply app/javascript/pages/app/components/app.jsx:39]: Not sure what you're asking him to do here. 🤔

**Comment by Eleara** [comment]
oops that was accidental

File: app/javascript/pages/app/icons/collapse_arrow-icon.svg (hunk 4 above)
Comment (KrzysztofMoszczynski) [reply app/javascript/pages/app/icons/collapse_arrow-icon.svg:1]: I just downloaded it from the internet

File: app/javascript/pages/app/styles/device-info.scss [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+    .location {
+      font-size: 12px;
+      color: #4f4f4f;
+    }
+    .collapse-arrow {
+      width: 15px;
+      height: 15px;
+      color: #4f4f4f;
+      cursor: pointer;
```
Comment (Eleara) [inline app/javascript/pages/app/styles/device-info.scss:27]: Does this color work? Because when I did it this way, it didn't work. Did you check that?

Comment (Eleara) [reply app/javascript/pages/app/styles/device-info.scss:27]: Try adding fill="#4f4f4f" to the svg file.

**Pull request merged**